portals/publisher/src/main/webapp/source/src/app/components/Shared/PoliciesUI/PoliciesExpansion.tsx

```diff
@@ -58,106 +58,192 @@ interface PoliciesExpansionSharedProps {
     PolicyDropzone: any;
 }
 
-const PoliciesExpansionShared: FC<PoliciesExpansionSharedProps> = ({
-    target,
-    verb,
-    allPolicies,
-    isChoreoConnectEnabled,
-    isAPILevelPolicy,
-    requestFlowPolicyList,
-    setRequestFlowPolicyList,
-    requestFlowDroppablePolicyList,
-    responseFlowPolicyList,
-    setResponseFlowPolicyList,
-    responseFlowDroppablePolicyList,
-    faultFlowPolicyList,
-    setFaultFlowPolicyList,
-    faultFlowDroppablePolicyList,
-    FlowArrow,
-    PolicyDropzone
-}) => {
+// Option 1: `listOriginatedFromCommonPolicies` and `isApiRevision` are provided
+interface PoliciesWithCommonProps extends PoliciesExpansionSharedBaseProps {
+    listOriginatedFromCommonPolicies: string[];
+    isApiRevision: boolean;
+}
+
+// Option 2: Neither `listOriginatedFromCommonPolicies` nor `isApiRevision` are provided
+interface PoliciesWithoutCommonProps extends PoliciesExpansionSharedBaseProps {
+    listOriginatedFromCommonPolicies?: undefined;
+    isApiRevision?: undefined;
+}
 
+// Combine the two using a union type
+type PoliciesExpansionSharedProps = PoliciesWithCommonProps | PoliciesWithoutCommonProps;
 
-    return (
-        <StyledAccordionDetails>
-            <Grid
-                spacing={2}
-                container
-                direction='row'
-                justifyContent='flex-start'
-                alignItems='flex-start'
-            >
-                <Grid item xs={12} md={12}>
-                    <Box className={classes.flowSpecificPolicyAttachGrid} data-testid='drop-policy-zone-request'>
-                        <Typography variant='subtitle2' align='left'>
-                            <FormattedMessage
-                                id='Apis.Details.Policies.PoliciesExpansion.request.flow.title'
-                                defaultMessage='Request Flow'
+const PoliciesExpansionShared: FC<PoliciesExpansionSharedProps> = (props) => {
```

Comment:
Destruct props

Reply:
relates to https://github.com/wso2/apim-apps/pull/809#discussion_r1901575381